neuralogic/core/constructs/rule.py

```diff
@@ -54,7 +55,10 @@ def __init__(self, head, body):
         if not isinstance(body, Iterable):
             body = [body]
 
-        self.body = list(body)
+        self.body = body
```

Comment:
There is a proposed syntax for lambdas, e.g.,:
```
R.a <= (b := R.b, c := R.c) >> F.tanh(b + c)
```

This will allow a lot of messy scenarios:
```
R.a <= (F.relu(b := R.b) * (c := R.c)) >> F.tanh(b + c)
```

Reply:
This need some more thinking. What if we have the following:

```
F.relu(R.a) <= (R.b + R.c) | [F.avg]
```
Which would correspond to:

```
(R.a <= (R.b, R.c)) | [F.avg, F.sum, F.relu]
```

Reply:
There will always be just one aggregation function, as it aggregates single values from all rules into one. Transformation on head could be stacked (e.g., multiple `tanh(relu(R.a))`, the same way could be stacked functions for relations in the body.

Reply:
The issue is readability - it's mixing computation with logic - but we already do that with soft negation hack (activation functions in the body is already possible)

Reply:
We also cannot do the following `F.tanh(b + c * c)` - that might be possible with some hacking?

```
F.relu(R.a) <= (R.b + (c  := R.c) * c) | [F.avg]
```